tests/e2e/tests/galley/galley_test.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package galley defines integration tests that validate working galley
+// functionality in context of a test Istio-enabled cluster.
+package galley
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/tests/e2e/framework"
+	"istio.io/istio/tests/util"
+)
+
+const (
+	// This namespace is used by default in all mixer config documents.
+	// It will be replaced with the test namespace.
+	templateNamespace = "istio-system"
+	yamlExtension     = "yaml"
+)
+
+type testConfig struct {
+	*framework.CommonConfig
+	configDir string
+}
+
+var (
+	tc *testConfig
+)
+
+func setTestConfig() error {
+	cc, err := framework.NewCommonConfig("galley_test")
+	if err != nil {
+		return err
+	}
+	tc = &testConfig{CommonConfig: cc}
+
+	// TODO - enforce test constraints
```

Comment:
@ozevren, checking flags is less than ideal but does succeed in guarding the test from running in the wrong environment. Ideally this could be written as `test.Required("galley", "cluster-wide")` at the test framework would ensure galley is running in a cluster-wide environment.